src/wp-includes/kses.php

```diff
@@ -2377,6 +2377,7 @@ function safecss_filter_attr( $css, $deprecated = '' ) {
 
 	$css = wp_kses_no_null( $css );
 	$css = str_replace( array( "\n", "\r", "\t" ), '', $css );
+	$css = WP_HTML_Decoder::decode_attribute( $css );
```

Comment:
What if the incoming string has encoded nefarious HTML in it? For example, what if it's this

```php
'background-image: url("&#60script&#62alert`1`&#60/script&#62")'
```

Result:
```
background-image: url("<script>alert`1`</script>")
``` 

Seems unsafe unless it gets re-encoded before returning.

@dmsnell @peterwilsoncc

Reply:
Hi @hellofromtonya - yes, thank you for noticing. I guess I didn't lean in enough into mentioning [re-encoding](https://github.com/WordPress/wordpress-develop/pull/6645#discussion_r1616440068) or [transforming back](https://github.com/WordPress/wordpress-develop/pull/6645#issuecomment-2389403312).

the root cause is whether we apply our logic on the kind of string that we think we are. for example, it doesn't help to look for `<script>` in a base-64-encoded string because it won't be there. we have to transform out of base-64, apply our logic, and transform back because something on the other end expects that base-64 (this is a hypothetical case, not meant to apply to this PR w.r.t. base-64).

it _is_ incredibly complicated because much of the code we deal with is agnostic to the encoding and escaping of strings they deal with. both sides are important, and I apologize for not asserting the re-encoding before sharing my earlier approval.

to your question, if the string has nefarious encoded HTML then it shouldn't matter, because a browser will not interpret HTML that's encoded as text, as long as we don't allow through a string that closes the attribute. when re-encoded, that content which looks like HTML will become CSS once again. we can mirror the argument when not decoding: what if unsafe HTML is hidden from our filters through clever escaping? there's an unlimited number of ways to represent any given character in HTML, so unless we perform actual parsing of the text, our checks won't be much better than if we applied them to a base-64-encoded string.

here are some summary points:

 - it seems like this code is specific to inline styles on a `style` attribute. that's important because `STYLE` contents do not transform. a `&lt;` in a `style` attribute is the text `<` but if found in `<style>&lt;</style>` it's the actual text `&lt;`.
 - we're presuming that this is extracted textually from a `style` attribute and not coming through the HTML API, which already decodes the attribute value.
 - we're sending this back out to the update the HTML attribute through an escaping function that will re-encode it

if any of these things are not true, then yes, we absolutely should resolve it.